### PR TITLE
feat(hover): Phase 3 - visual hierarchy and documentation sections (#727)

### LINKED ISSUE #727: [hover] Phase 3: Visual hierarchy and documentation sections
## Problem
Hover documentation currently displays as flat text without visual hierarchy. Need better organization of documentation sections.

## Context
Part of Hover Redesign (#706). Phases 1-2 completed in PRs #713 and #716.

## Proposed Approach
- Add section headers for Parameters, Returns, Throws, See Also
- Prominent deprecation badges/warnings with replacement hints
- `@since` and `@author` badges
- Visual separators between sections
- Smart truncation with "show more" hints

## References
- Parent issue: #706
- Plan: `.plan/hover-redesign.md`
- PRs: #713, #716

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary

This PR implements **Phase 3** of the hover redesign (#727), adding visual hierarchy and improved documentation formatting to make hover content more scannable and professional.

#### 1. Deprecation Warnings at Top with Visual Prominence
- Moved deprecation notices to the **top** of hover content
- Uses blockquote format with warning emoji: `> ⚠️ **Deprecated**`
- Ensures deprecated APIs are immediately visible before reading further

#### 2. Section Headers for Better Organization
- Added markdown headers (`####`) for all documentation sections:
  - **Parameters**, **Returns**, **Throws**, **See Also**
- Improves scannability and creates clear visual hierarchy
- Users can quickly jump to relevant sections

#### 3. Smart Parameter Formatting
- **Table format** for methods with 2+ parameters:
  - Includes Name, Type, Description columns
  - Better for complex method signatures with many parameters
- **Inline format** for single parameters:
  - Cleaner, more concise display
  - Uses em dash separator: `**param** — description`

#### 4. Metadata Footer
- Display `@since` and `@author` in italics at bottom
- Separate multiple metadata items with bullet separator (·)
- Add horizontal rule before footer for visual separation
- Keeps metadata visible but not distracting

#### 5. Visual Separators
- Add horizontal rules (`---`) between major sections
- Improves document structure and readability
- Creates clear boundaries between content, documentation, and metadata

### Example Output

**Before (Phase 2):**
```markdown
Calculates the sum of two numbers.

**Parameters:**
- `x`: the first number
- `y`: the second number

**Returns:** the sum

**Deprecated**: Use Calculator.add() instead.
```

**After (Phase 3):**
```markdown
> ⚠️ **Deprecated**: Use Calculator.add() instead.

Calculates the sum of two numbers.

---

#### Parameters
| Name | Type | Description |
| --- | --- | --- |
| `x` |  | the first number |
| `y` |  | the second number |

#### Returns
the sum

---

*@since 1.0* · *@author John Doe*
```

## Testing Approach

Followed **TDD** (Test-Driven Development):

1. ✅ Wrote 11 new comprehensive tests covering all Phase 3 features
2. ✅ Ran tests to verify they failed initially (red)
3. ✅ Implemented the functionality to make tests pass (green)
4. ✅ Updated existing tests to match new format
5. ✅ All 23 DocFormatter tests passing

### Test Coverage

New tests added:
- Deprecation warning at top with visual prominence
- Multiple params use table format with section header
- Single param uses inline format
- Return doc shows section header
- Throws shows section header
- Metadata shows footer with separators
- Visual separators between major sections
- See Also shows section header
- No deprecation warning when not deprecated
- Footer formatting without leading separator

## Quality Checklist

- [x] Tests written FIRST (TDD)
- [x] All new tests pass
- [x] Existing tests updated and pass
- [x] Spotless applied
- [x] No breaking changes
- [x] Follows project conventions

## Related Issues

Closes #727

## Next Steps

Phase 4 (if planned): Integration with signature formatting for complete hover experience

---

> [!NOTE]
> Introduces Phase 3 hover markdown formatting improvements in `DocFormatter` and expands tests to validate the new structure.
> 
> - Deprecation moved to the top as a blockquote with emoji (`> ⚠️ **Deprecated** ...`)
> - Adds `####` headers for `Parameters`, `Returns`, `Throws`, and `See Also`
> - Smart parameters: table for multiple params, inline with em dash for single param
> - Inserts `---` separators between content, doc sections, and footer
> - Metadata footer renders `@since`/`@author` in italics joined by ` · `
> - Updates existing tests and adds comprehensive new tests to cover all Phase 3 behaviors
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 02b87c91a2e5f148319ab04e6be7d8615b1ef353. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Deprecation warnings displayed prominently at the top of documentation
  * Structured documentation sections with clear headers (Parameters, Returns, Throws, See Also)
  * Parameters formatted as tables for multiple entries or inline for single entries
  * Metadata footer displaying since and author information
  * Enhanced visual separation between documentation sections

* **Tests**
  * Expanded test coverage for new formatting scenarios and edge cases

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: groovy-lsp/src/test/kotlin/com/github/albertocavalcante/groovylsp/documentation/DocFormatterTest.kt [hunk 1]
```diff
@@ -174,7 +185,191 @@ class DocFormatterTest {
...
+            summary = "Simple method.",
+            since = "2.0",
+        )
+
+        val result = DocFormatter.formatAsMarkdown(doc)
+
+        // Should have metadata but check separator placement
+        assertTrue(result.contains("*@since 2.0*"))
+        assertTrue(result.contains("---"))
     }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-lsp/src/test/kotlin/com/github/albertocavalcante/groovylsp/documentation/DocFormatterTest.kt]: Fixed in 02b87c91. Renamed test to clarify it checks WITH content, and added new test for metadata-only case that properly asserts no separator.

File: groovy-lsp/src/main/kotlin/com/github/albertocavalcante/groovylsp/documentation/DocFormatter.kt [hunk 2]
```diff
@@ -30,38 +35,68 @@ object DocFormatter {
                 text(InlineTagRenderer.render(doc.description))
             }
 
-            // Add deprecated notice
-            if (doc.deprecated.isNotBlank()) {
-                text("**Deprecated**: ${InlineTagRenderer.render(doc.deprecated)}")
+            // Phase 3: Add visual separator before documentation sections if we have content above
+            val hasContentAbove = doc.summary.isNotBlank() || doc.description.isNotBlank()
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-lsp/src/main/kotlin/com/github/albertocavalcante/groovylsp/documentation/DocFormatter.kt]: Fixed in 02b87c91. Included doc.deprecated.isNotBlank() in hasContentAbove calculation to ensure separator appears between deprecation and doc sections.
